tests/trace_service.py

```diff
@@ -31,14 +31,25 @@ def get_span(self):
 
 
 @pytest.fixture(scope="module")
-def trace_service(pytestconfig, logger):
+def trace_service(pytestconfig, testdir, logger, certs):
     server = grpc.server(concurrent.futures.ThreadPoolExecutor())
     trace_service = TraceService()
     trace_service_pb2_grpc.add_TraceServiceServicer_to_server(
         trace_service, server
     )
     listen_addr = f"127.0.0.1:{24317 if pytestconfig.option.otelcol else 14317}"
     server.add_insecure_port(listen_addr)
+    if pytestconfig.option.otelcol is None:
+        creds = grpc.ssl_server_credentials(
+            [
+                (
+                    (testdir / "localhost.key").read_text().encode(),
```

Comment:
Given, we generate cert/key in memory, perhaps, we shouldn't read them from files here. We can make `certs` fixture into a dict `cert` (as there is only one for now), so in-memory representation will be easily available, e.g.

```
@pytest.fixture(scope="module")
def cert(testdir):
    cert = self_signed_cert("localhost") # { chain: "...", key: "..." }
    ...write cert.chain to file...
    ...write cert.key to file...
    yield cert
```

Reply:
Make sense. Fixed.

Reply:
Not quite, although tuple might be fine for now.